docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+---
+title: Why Gatsby Uses GraphQL
+---
+
+A common question about Gatsby is, “Why does Gatsby use GraphQL? Doesn’t it generate static files?”
+
+Without providing some context, it can seem like GraphQL is overkill for something like Gatsby. In this document, you’ll see what problems arise when creating pages, and how those problems can be solved using GraphQL.
+
+## Create a page without any data
+
+For any kind of pages that aren’t directly created in `src/pages/`, you’ll need Gatsby’s [`createPages` Node API](/docs/node-apis/#createPages) to create pages programmatically.
+
+All that’s required to create a page is a `path` where it should be created and the component that should be rendered there.
+
+For example, if you had the following component:
+
+```jsx:title=src/templates/no-data.js
+import React from "react"
+
+const NoData = () => (
+  <section>
+    <h1>This Page Was Created Programmatically</h1>
+    <p>
+      No data was required to create this page — it’s just a React component!
```

Comment:
This text is cut off in the code example. Not a huge deal, but I thought I'd pass that along!

Reply:
Cut off? Or scrolling?

Reply:
I meant scrolling, sorry!